### PR TITLE
New object structure 2024-11-27

### PR DESCRIPTION
**Pull request by EstevaoCalera**
The 2024-11-27 update brings some structure changes:

* 'ShowBackground = true' on every annotation

* 'GrenadeType' on main annotations to specify which grenade it refers to

### CODE REVIEW

**Comment by ReneRebsdorf** [comment]
Awesome, thank you!

You were aster than me, just put it on issuetracker before going to bed
Will review and merge this later today.

File: de_ancient.txt [hunk 1]
```diff
@@ -84,13 +89,15 @@
 			FontSize = 75
 			FadeInDist = 50.0
 			FadeOutDist = -1.0
+			ShowBackground = true
```
Comment (ReneRebsdorf) [inline de_ancient.txt:92]: I need to check whether the background is hidden when no text is present. If it needs to be false in that case I will write a test for it

**Review by ReneRebsdorf — CHANGES_REQUESTED** [review]
Looks good, one comment to verify i game, if you already tested this we can resolve and merge

File: de_ancient.txt (hunk 1 above)
Comment (EstevaoCalera) [reply de_ancient.txt:92]: When there is no text, the background doesn't show either. In the following image, the "standing instructions" are empty and nothing is shown, even though ShowBackground = true

image

**Comment by EstevaoCalera** [comment]
I have added two commits to fix some errors:

* Removed unnecessary GrenadeType = "smoke" from text annotations

* Changed the file location to fix the issue #22

**Review by ReneRebsdorf — APPROVED** [review]

**Review by ReneRebsdorf — CHANGES_REQUESTED** [review]
`.github/workflows/main.yml` needs updating as the zip file should be immediately extractable to the csgo folder
so we should just zip the local folder to local.zip for the easiest user experience I think.

This affects :16 and :20

I can do this later today for this PR, and I think we are good to merge then :)

**Comment by EstevaoCalera** [comment]
I would love to help you with that workflow, but I don't know exactly how to test this Github Actions. I've never used it before.

How can I test it locally? Is there any VSCode extensions?

**Comment by ReneRebsdorf** [comment]
> I would love to help you with that workflow, but I don't know exactly how to test this Github Actions. I've never used it before.
> 
> How can I test it locally? Is there any VSCode extensions?

I have never looked into it, but I've heard https://github.com/nektos/act might be what you are looking for.
But I've mostly just relied on making branches, or even other repos, to test my builds and verify them before merging.

In other words, I just use the actual github workflows to verify their functionality. It would be great to be able to run them locally as I much prefer testing before commit+push.

**Review by ReneRebsdorf — APPROVED** [review]

**Pull request merged**